core/timeout_handler.py

```
@@ -0,0 +1,252 @@
+            
+            try:
+                return await asyncio.wait_for(
+                    func(*args, **kwargs),
+                    timeout=seconds
+                )
+            except asyncio.TimeoutError:
+                elapsed = (datetime.now() - start_time).total_seconds()
+                logger.warning(
+                    f"Async timeout: {op_name} exceeded {seconds}s (elapsed: {elapsed:.2f}s)"
+                )
+                raise TimeoutError(op_name, seconds, start_time)
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)